### PR TITLE
Add Bin without span to New Query Pipeline

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- Now `bin` command without span options is supported in the query pipeline.

# Testing
- Tested by running some bin without span queries and verified by running the same queries through UI.
```
curl --location 'http://localhost:5122/api/search' \
--header 'Content-Type: application/json' \
--data '{
    "state": "query",
    "searchText": "city=boston | bin bins=20 http_status AS http_status_bin",
    "startEpoch": "now-7d",
    "endEpoch": "now",
    "indexName": "*",
    "from": 0,
    "queryLanguage": "Splunk QL"
}'
```

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/query/processor/bincommand.go [hunk 1]
```diff
@@ -89,14 +111,35 @@ func (p *binProcessor) Process(iqr *iqr.IQR) (*iqr.IQR, error) {
...
+	p.options.BinSpanOptions = binSpanOptions
 }
 
 func (p *binProcessor) Cleanup() {
-	// Nothing to do.
+	p.initializedMinMax = false
+	p.minVal = 0
+	p.maxVal = 0
+	p.secondPass = false
+	p.spanError = nil
```
[the remark this answers is not in the document]
Comment (nkunal) [reply pkg/segment/query/processor/bincommand.go:145]: if we keep this, does it set a precendence that in cleanup code of all the other (or new) cmds that we add, a new developer would need to do such things ? 
Or should we just add a comment here why we needed to reset these vars ? like are they going to be reused after a cleanup is being called ?

**Review by nkunal — COMMENTED** [review]
do we have any UTs for the helper methods like

**Review by nkunal — COMMENTED** [review]
do we have any UTs for these methods : 
1. findSpan
2. getSecsFromMinSpan
3. findEstimatedTimeSpan
4.

**Comment by Macbeth98** [comment]
> do we have any UTs for these methods :
> 
> 1. findSpan
> 2. getSecsFromMinSpan
> 3. findEstimatedTimeSpan

@nkunal I have added UTs for `findSpan` function, which calls the other two functions `getSecsFromMinSpan` and `findEstimatedTimeSpan`.

File: pkg/segment/query/processor/bincommand.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndrewHess) [reply pkg/segment/query/processor/bincommand.go:145]: Ok in that case let's remove this, so we don't set a bad precedent. The `Cleanup()` method was originally for certain commands that might do things like open files (e.g., if sort has a super high limit, or inputlookup keeps the file open across calls to Process)

**Review by AndrewHess — COMMENTED** [review]
Looks good, just remove the Cleanup() stuff like we talked about

**Comment by Macbeth98** [comment]
> Looks good, just remove the Cleanup() stuff like we talked about

Done.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**